hydra-cluster/src/HydraNode.hs

```diff
@@ -281,9 +282,9 @@ withHydraNode ::
 withHydraNode tracer chainConfig workDir hydraNodeId hydraSKey hydraVKeys allNodeIds pparams action = do
   withLogFile logFilePath $ \logFileHandle -> do
     withHydraNode' chainConfig workDir hydraNodeId hydraSKey hydraVKeys allNodeIds pparams (Just logFileHandle) $ do
-      \_ _ processHandle -> do
+      \_ err processHandle -> do
         race
-          (checkProcessHasNotDied ("hydra-node (" <> show hydraNodeId <> ")") processHandle)
+          (checkProcessHasNotDied ("hydra-node (" <> show hydraNodeId <> ")") processHandle err)
```

Comment:
Should: not weaken the interface of `withHydraNode'` by keeping `Handle` there and using `(Just err)` here
```suggestion
          (checkProcessHasNotDied ("hydra-node (" <> show hydraNodeId <> ")") processHandle $ Just err)
```